pkg/apis/autoscaling/validation/validation.go

```diff
@@ -442,6 +442,14 @@ func validateMetricTarget(mt autoscaling.MetricTarget, fldPath *field.Path) fiel
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("type"), mt.Type, "must be either Utilization, Value, or AverageValue"))
 	}
 
+	if mt.Type == autoscaling.ValueMetricType && mt.Value == nil && mt.AverageValue != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("value"), mt.Value, "must be set for metric target type Value"))
+	}
+
+	if mt.Type == autoscaling.AverageValueMetricType && mt.AverageValue == nil && mt.Value != nil {
```

Comment:
We only care that when Type is AverageValue, averageValue is specified and when type is Value, value is specified. I think we can ignore if Value is set or not in this specific check

Reply:
The only value of additionally checking if Value is set or not for type AverageValue is to show a more helpful message saying , `you set value but instead you need to set averageValue`

Reply:
Thanks for taking a look! I've made the requested change to only care about the mapping we're looking for, and removed some now redundant tests as they're calling `validateMetricTarget` beforehand. Git made the changes looks kind of weird with the deletes, here's the diff if that's any help https://gist.github.com/Lukasz-AWS/4317e9e556d052d0e2e5f3af51a77cba